### PR TITLE
Bounding boxes crop migration

### PR DESCRIPTION
**Pull request by index-git**
Part of issue #367 

- [x] Tests
- ~~Layman Test Client~~
- [x] Changlelog
- ~~Documentation~~

Optionally
- Add dependency to doc/dependencies.md

### CODE REVIEW

File: src/layman/upgrade/upgrade_v1_14.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+from layman.layer.qgis import wms as qgis_wms
+from layman.map import MAP_TYPE
+from layman.map.filesystem import thumbnail as map_thumbnail
+from layman.map.micka import soap as map_micka_soap
+
+logger = logging.getLogger(__name__)
+DB_SCHEMA = settings.LAYMAN_PRIME_SCHEMA
+
+
+def crop_bbox():
```
Comment (index-git) [inline src/layman/upgrade/upgrade_v1_14.py:20]: Add this migration to migration list in `src/layman/upgrade/__init__.py`

File: src/layman/upgrade/upgrade_v1_14.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+where st_xMin(p.bbox) < -20026376.39
+   or st_yMin(p.bbox) < -20048966.10
+   or st_xMax(p.bbox) > 20026376.39
+   or st_yMax(p.bbox) > 20048966.10
+;'''
+    publications = db_util.run_query(query, settings.LAYMAN_DEFAULT_OUTPUT_BBOX)
+    for workspace, publ_type, publication, xmin, ymin, xmax, ymax in publications:
+        info = layman_util.get_publication_info(workspace, publ_type, publication, context={'keys': ['style_type', 'file', 'uuid'], })
+        bbox = (xmin, ymin, xmax, ymax)
+        publications_util.set_bbox(workspace, publ_type, publication, bbox)
```
Comment (jirik) [inline src/layman/upgrade/upgrade_v1_14.py:40]: Use crop logic & SQL statement directly

File: src/layman/upgrade/upgrade_v1_14.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            gs_wms.clear_cache(workspace)
+
+            # Thumbnail
+            layer_thumbnail.generate_layer_thumbnail(workspace, publication)
+
+            # Micka soap
+            layer_micka_soap.patch_layer(workspace, publication, metadata_properties_to_refresh=['extent'])
+        elif publ_type == MAP_TYPE:
+            # Thumbnail
+            map_thumbnail.generate_map_thumbnail(workspace, publication, editor=None)
```
Comment (jirik) [inline src/layman/upgrade/upgrade_v1_14.py:68]: Remove this line

File: src/layman/upgrade/upgrade_v1_14.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            elif style_type == 'qml':
+                qgis_wms.save_qgs_file(workspace, publication)
+                gs_util.patch_wms_layer(geoserver_workspace, publication, auth=settings.LAYMAN_GS_AUTH, bbox=bbox)
+            gs_wms.clear_cache(workspace)
+
+            # Thumbnail
+            layer_thumbnail.generate_layer_thumbnail(workspace, publication)
+
+            # Micka soap
+            layer_micka_soap.patch_layer(workspace, publication, metadata_properties_to_refresh=['extent'])
```
Comment (jirik) [inline src/layman/upgrade/upgrade_v1_14.py:65]: Assert that all bboxes are same using internal metadata comparison

File: CHANGELOG.md [hunk 5]
```diff
@@ -30,6 +30,7 @@
 ### Migrations and checks
 #### Schema migrations
 #### Data migrations
+- All bounding boxes are cropped to not exceed (-20026376.39, -20048966.10, 20026376.39, 20048966.10) (in EPSG:3857)
```
Comment (jirik) [inline CHANGELOG.md:33]: All bounding boxes are cropped not to exceed world boundaries ([-20026376.39, -20048966.10, 20026376.39, 20048966.10] in EPSG:3857).

**Review by jirik — CHANGES_REQUESTED** [review]

**Review by jirik — DISMISSED** [review]

**Review by jirik — APPROVED** [review]

**Pull request merged**